Bundle Source Link packages

tmat opened the pull request:
Source Link targets are included automatically unless the project has already a `PackageReference` to any Source Link package.

Property `SuppressImplicitGitSourceLink` can be used to disable automatic inclusion of Source Link.

tmat committed:
Bundle Source Link packages

dsplaisted commented on src/Assets/TestProjects/SourceLinkTestApp/SourceLinkTestApp.csproj:
```diff
@@ -0,0 +1,8 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <RestorePackagesPath Condition=" '$(RestorePackagesPath)' == '' ">$(MSBuildThisFileDirectory)..\..\..\artifacts\.nuget\packages</RestorePackagesPath>
```
Looking at the tests, it looks like this project is copied to the temp folder, so the relative path will be wrong anyway.

dsplaisted commented on src/Tests/Microsoft.NET.Build.Tests/SourceLinkTests.cs:
```diff
@@ -0,0 +1,238 @@
+        }
+
+        /// <summary>
+        /// When creating a new repository locally we want the build to work and not report warnings even before the remote is set.
```
This description (that the build shouldn't report warnings) seems to contradict the assertion (that it should include a warning).

tmat replied:
Ah, forgot to update the assert. Was testing it manually.

dsplaisted reviewed (approved):
I haven't looked at the code for each sourcelink package, so I'm assuming those are good to be integrated into the SDK.

Once this flows into dotnet/installer, we should check to see whether the sourcelink SDKs are appropriately crossgen'd.

tmat commented on src/Layout/redist/targets/RestoreDependency.proj:
```diff
@@ -1,4 +1,4 @@
-<Project ToolsVersion="15.0" DefaultTargets="EnsureDependencyRestored;CopySdkToOutput">
+﻿<Project ToolsVersion="15.0" DefaultTargets="PrepareBundledDependencyProps;EnsureDependencyRestored;CopySdkToOutput">
```
Removed from dependencies

tmat committed:
Feedback

tmat committed:
Add multitargeting tests.

tmat committed:
Move SourceLink targets imports out of Microsoft.NET.Sdk.Common.targets.
Microsoft.NET.Sdk.Common.targets is imported from Solution.targets but corresponding props file isn't, which breaks the targets.

tmat committed:
C++

tmat committed:
Fix

tmat committed:
Fix

tmat committed:
Fix versions

tmat committed:
Feedback
